app/services/action_runner.py

```diff
@@ -47,35 +46,31 @@ async def execute_action(integration_id: str, action_id: str, config_overrides:
             content=jsonable_encoder({"detail": message}),
         )
 
-    # If "config" variable present in the request, we bypass the integration/config retrieval from the portal
-    if config_data:
-        config = config_data
-    else:
-        # Look for the configuration of the action being executed
-        action_config = find_config_for_action(
-            configurations=integration.configurations,
-            action_id=action_id
+    # Look for the configuration of the action being executed
+    action_config = find_config_for_action(
+        configurations=integration.configurations,
+        action_id=action_id
+    )
+    if not action_config and not config_overrides:
+        message = f"Configuration for action '{action_id}' for integration {str(integration.id)} " \
+                  f"is missing. Please fix the integration setup in the portal or include a config to override."
```

Comment:
I like this, as it'll accommodate both the interactive executed actions as well as sub-task actions.